ui-tests/tools/com.liferay.ide.functional.modules.ext/com.liferay.ide.functional.modules.ext.wizard/com.liferay.ide.functional.modules.ext.wizard.validation/src/com/liferay/ide/functional/modules/ext/validation/tests/ValidationModulesExtTests.java

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.functional.modules.ext.validation.tests;
+
+import com.liferay.ide.ui.liferay.SwtbotBase;
+import com.liferay.ide.ui.liferay.support.project.ProjectSupport;
+import com.liferay.ide.ui.liferay.support.project.ProjectsSupport;
+import com.liferay.ide.ui.liferay.util.ValidationMsg;
+import com.liferay.ide.ui.swtbot.util.StringPool;
+
+import java.io.File;
+
+import org.eclipse.core.runtime.Platform;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Rui Wang
+ */
+public class ValidationModulesExtTests extends SwtbotBase {
+
+	@Test
+	public void checkExtWithoutWorkspace() {
+		wizardAction.openNewLiferayModulesExtWizard();
+
+		validationAction.assertEquals(PLEASE_ENTER_A_PROJECT_NAME, wizardAction.getValidationMsg(4));
+
+		wizardAction.newModulesExt.prepare(project.getName());
+
+		validationAction.assertEquals(
+			WE_RECOMMEND_LIFERAY_GRADLE_WORKSPACE_TO_DEVELOP_MODULE_EXT_PROJECT, wizardAction.getValidationMsg(4));
+
+		validationAction.assertTextEquals(StringPool.BLANK, wizardAction.newModulesExt.origialModuleName());
+
+		validationAction.assertTextEquals(StringPool.BLANK, wizardAction.newModulesExt.origialModuleVersion());
+
+		validationAction.assertEnabledFalse(wizardAction.getFinishBtn());
+
+		wizardAction.cancel();
+	}
+
+	@Test
+	public void checkExtWithValidWorkspace() {
+		wizardAction.openNewLiferayWorkspaceWizard();
+
+		wizardAction.newLiferayWorkspace.prepareGradleWithIndexSources(project.getName());
+
+		wizardAction.finish();
+
+		jobAction.waitForNoRunningJobs();
+
+		wizardAction.openNewLiferayModulesExtWizard();
+
+		validationAction.assertTextEquals(StringPool.BLANK, wizardAction.newProject.projectName());
```

Comment:
delete this, or you should check validation message, not checking projectname is empty here